### PR TITLE
Destroy all connections beforehand, make new ones from the list

### LINKED ISSUE #41: Initial checkin response handling
When the engine checks in with the router, the router should return to it a listing of all known services and their models. The engine should then update its own database as necessary.

Currently it throws the router's response into the trash except to see whether the response's status code was a 200.

### PR DESCRIPTION
**Pull request by dfaulken**
Still needs tests.

Closes #41.

### CODE REVIEW

File: lib/generators/install/templates/microservices_engine.rb [hunk 1]
```diff
@@ -1,34 +1,69 @@
...
+          raise IOError, "No router configuration file found at #{filename}"
+        end
+      end
 
-  service_name = config_data.fetch 'name', &report_missing_config_key
-  service_url = config_data.fetch 'uri', &report_missing_config_key
-  router_url = config_data.fetch 'router_uri', &report_missing_config_key
-  security_token = config_data.fetch 'security_token', &report_missing_config_key
+      def check_in_with_router(options = {})
+        if options.key? :using
```
Comment (Anbranin) [inline lib/generators/install/templates/microservices_engine.rb:27]: Just curious, why this? Why not just pass it config_data directly?

Comment (dfaulken) [reply lib/generators/install/templates/microservices_engine.rb:27]: I'm not that attached to it, just a stylistic choice I made to help indicate that the config data is going through some interpolation before getting passed to the router.

**Comment by Anbranin** [comment]
TESTED

File: spec/microservices_engine/initializer_spec.rb [hunk 2]
```diff
@@ -26,16 +28,41 @@
     end
     context 'otherwise' do
       it 'checks for a config file and parses it as YAML' do
+        stub_request(:post, "http://example.com/services/register")
```
Comment (dfaulken) [inline spec/microservices_engine/initializer_spec.rb:31]: Nice, what library provides this functionality?

File: spec/microservices_engine/initializer_spec.rb [hunk 3]
```diff
@@ -26,16 +28,41 @@
...
           .with 'apples'
         expect(YAML).to receive(:load_file).with('apples')
           .and_return config_data
         call
       end
-      it 'checks in with the router using the data from the config file'
-      it "raises an exception if the router's response code was not 200"
-      it 'updates the connections database using the response from the router'
+      it 'checks in with the router using the data from the config file' do
+        allow(described_class).to receive(:check_for_config_file!) { nil }
```
Comment (dfaulken) [inline spec/microservices_engine/initializer_spec.rb]: Does `and_return` not work here?

File: spec/microservices_engine/initializer_spec.rb [hunk 4]
```diff
@@ -26,16 +28,41 @@
...
+        allow(YAML).to receive(:load_file) { config_data }
+        allow(JSON).to receive(:parse) { response_data }
+        response = Struct.new(:code, :body).new('200', 'body')
```
Comment (dfaulken) [inline spec/microservices_engine/initializer_spec.rb]: This is what test `double`s are for.

```ruby
double code: 200, body: 'body'
```

**Review by dfaulken — COMMENTED** [review]
A few things.

File: spec/microservices_engine/initializer_spec.rb (hunk 2 above)
Comment (Anbranin) [reply spec/microservices_engine/initializer_spec.rb:31]: Webmock!

File: spec/microservices_engine/initializer_spec.rb (hunk 3 above)
Comment (Anbranin) [reply spec/microservices_engine/initializer_spec.rb]: The documentation for the version of RSpec that we have made no mention of it, it's listed under "old syntax." However it does not throw any errors or deprecation warnings, so I can use it instead if that's preferred.

File: spec/microservices_engine/initializer_spec.rb (hunk 4 above)
Comment (Anbranin) [reply spec/microservices_engine/initializer_spec.rb]: Oh

File: spec/microservices_engine/initializer_spec.rb (hunk 3 above)
Comment (Anbranin) [reply spec/microservices_engine/initializer_spec.rb]: Of course, 'and_return' is totally used elsewhere in there, so I don't know why I didn't just stay consistent. 😛

**Review by Anbranin — APPROVED** [review]

File: lib/generators/install/install_generator.rb [hunk 5]
```diff
@@ -6,6 +6,6 @@ def setup_initializer
     template 'mse_router_info.yml', 'config/mse_router_info.yml'
     copy_file 'microservices_engine.rb',
               'config/initializers/microservices_engine.rb'
-    generate 'migration', 'create_microservices_engine_connections', 'name:string', 'object:string', 'url:string'
+    generate 'migration', 'create_microservices_engine_connections name:string object:string url:string'
```
Comment (Anbranin) [inline lib/generators/install/install_generator.rb]: We should move onto making a migration template instead, probably. This doesn't seem to work.

Comment (dfaulken) [reply lib/generators/install/install_generator.rb]: Agreed. I've received no response on SO, even with a rep bounty.

**Comment by Anbranin** [comment]
If this is all set, I'm going to merge and then work on a migration template.

**Pull request merged**